### ISSUE TITLE
Can't get rid of Syntax Error: Lexer: Undefined token

### ISSUE THREAD

**Issue by thery** [coq-community/vscoq#644]
Sometimes I can't get rid of a lexer problem. The expression is valid but it does not parse. 
Here is a way to reproduce it (not 100% sure it works)
I start with the code.
```coq
Variables ph pl : nat.
Lemma foo : (ph + pℓ − 0) <= 0.
```
it indicates an error on "0" now if I edit to remove the non ascii to get  (`ℓ −`) and get the correct code
```coq
Lemma foo : (ph + pl - 0) <= 0.
```
the lexer error remains

**Comment by rtetley** [comment]
I'm able to reproduce with the example. This is probably something to do with errors not being properly updated in the document state.

**Comment by thery** [comment]
I have a simpler version : 
Start with 
`Lemma sF : True.`

then add a `'` (I am in compose mode so I have to type ` then space)

`Lemma s'F : True.`

try to execute : Lexical error

Reload window, it works.

**Comment by rtetley** [comment]
Closed by #664
